.github/workflows/course-schedule-comment.yml

```diff
@@ -0,0 +1,68 @@
+# Based on https://securitylab.github.com/research/github-actions-preventing-pwn-requests/
+#
+# This runs with the full repo permissions, but checks out the upstream branch, not the PR.
+# Do not run arbitrary code from the PR here!
+name: Comment PR with Course Schedule
+on:
+  workflow_run:
+    workflows: ["Generate Course Schedule"]
+    types: [completed]
+
+jobs:
+  upload:
+    runs-on: ubuntu-latest
+    if: >
+      github.event.workflow_run.event == 'pull_request' &&
+      github.event.workflow_run.conclusion == 'success'
+    steps:
+      - name: "Checkout"
+        uses: actions/checkout@v4
+
+      - name: "Setup Rust cache"
+        uses: ./.github/workflows/setup-rust-cache
+
+      - name: "Generate Schedule on upstream branch"
+        run: |
+          mkdir -p ./course-schedule
+          cargo run -p mdbook-course --bin course-schedule > upstream-schedule
+
+      - name: "Download artifact from PR workflow"
+        uses: actions/github-script@v7.0.1
+        with:
+          script: |
+            var artifacts = await github.rest.actions.listWorkflowRunArtifacts({
+               owner: context.repo.owner,
+               repo: context.repo.repo,
+               run_id: ${{github.event.workflow_run.id }},
+            });
+            var matchArtifact = artifacts.data.artifacts.filter((artifact) => {
+              return artifact.name == "course-schedule"
+            })[0];
+            var download = await github.rest.actions.downloadArtifact({
+               owner: context.repo.owner,
+               repo: context.repo.repo,
+               artifact_id: matchArtifact.id,
+               archive_format: 'zip',
+            });
+            var fs = require('fs');
+            fs.writeFileSync('${{github.workspace}}/course-schedule.zip', Buffer.from(download.data));
+
+      - name: "Unzip artifact"
+        run: unzip course-schedule.zip
+
+      - name: "Comment on PR if schedules differ"
+        uses: actions/github-script@v7.0.1
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            var fs = require('fs');
+            var upstream = fs.readFileSync('upstream-schedule').toString();
+            var schedule = fs.readFileSync('schedule').toString();
+            if (upstream != schedule) {
+              await github.rest.issues.createComment({
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                issue_number: context.payload.workflow_run.pull_requests[0].number,
```

Comment:
This might be flaky as I discovered. In `mdbook-i18n-helpers` I fixed it by https://github.com/google/mdbook-i18n-helpers/pull/168

According to my testing, the root cause of the issue is that `context.payload.workflow_run.pull_requests` array is not always populated. Some discussion on this issue https://github.com/orgs/community/discussions/25220

Reply:
Yes, debugging them is very cumbersome. It is possible to create a PR against your own fork though. That's what I did in my fork https://github.com/kdarkhan/mdbook-i18n-helpers/pulls?q=is%3Apr+is%3Aclosed

Reply:
Yeah, that's how I was testing -- but I think it doesn't accurately simulate the permissions of a cross-owner PR.